src/IO/Resample/Resample.cxx

```diff
@@ -0,0 +1,160 @@
+#include "itkBinShrinkImageFilter.h"
+#include "itkVectorImage.h"
+#include "itkResampleImageFilter.h"
+#include "itkLabelImageGenericInterpolateImageFunction.h"
+#include "itkLinearInterpolateImageFunction.h"
+#include "itkImageRegionSplitterSlowDimension.h"
+#include "itkExtractImageFilter.h"
+#include "itkRGBPixel.h"
+#include "itkRGBAPixel.h"
+#include "itkVectorImage.h"
+#include "itkOffset.h"
+#include "itkVector.h"
+#include "itkPoint.h"
+#include "itkCovariantVector.h"
+#include "itkSymmetricSecondRankTensor.h"
+#include "itkDiffusionTensor3D.h"
+#include <complex>
+#include "itkFixedArray.h"
+#include "itkArray.h"
+#include "itkMatrix.h"
+#include "itkVariableLengthVector.h"
+#include "itkVariableSizeMatrix.h"
+#include <fstream>
+#include "itkPipeline.h"
+#include "itkInputImage.h"
+#include "itkOutputImage.h"
+#include "itkOutputTextStream.h"
+#include "itkSupportInputImageTypes.h"
+
+template <typename TImage>
+int ResampleLabelImage(itk::wasm::Pipeline &pipeline, itk::wasm::InputImage<TImage> &inputImage)
+{
+  using ImageType = TImage;
+
+  pipeline.get_option("InputImage")->required();
+
+  using OutputImageType = itk::wasm::OutputImage<ImageType>;
+  OutputImageType outputImage;
+  pipeline.add_option("OutputImage", outputImage, "Output image")->required();
+
+  std::vector<unsigned int> outSize;
+  pipeline.add_option("outSize", outSize, "New image size for each direction")->expected(2, 3)->delimiter(',');
+
+  unsigned int maxTotalSplits = 1;
+  pipeline.add_option("-m,--max-total-splits", maxTotalSplits, "Maximum total splits when processed in parallel");
+
+  unsigned int split = 0;
+  pipeline.add_option("-s,--split", split, "Current processed split");
+
+  itk::wasm::OutputTextStream numberOfSplitsStream;
+  auto numberOfSplitsStreamOption = pipeline.add_option("--number-of-splits", numberOfSplitsStream, "Number of splits");
+
+  ITK_WASM_PARSE(pipeline);
+
+  auto inImage = inputImage.Get();
+
+  using ResampleFilterType = itk::ResampleImageFilter<ImageType, ImageType>;
+  auto resampleFilter = ResampleFilterType::New();
+
+  using InterpolatorType = itk::LabelImageGenericInterpolateImageFunction<ImageType, itk::LinearInterpolateImageFunction>;
+  auto interpolator = InterpolatorType::New();
+  resampleFilter->SetInterpolator(interpolator);
+
+  resampleFilter->SetInput(inImage);
+  resampleFilter->SetOutputParametersFromImage(inImage);
```

Comment:
We want to specify the resampling grid parameters as arguments, i.e. `origin`, `spacing`, `direction`, `size`, and get these from the fixed image / intensity image, example:

https://examples.itk.org/src/filtering/imagegrid/resampleanimage/documentation?highlight=resampleimagefilter

Reply:
Feeding those 4 grid parameter to ResampleImageFilter now.